packages/webcert/src/components/icf/IcfDropdown.tsx

```diff
@@ -0,0 +1,151 @@
+    <>
+      <CustomButton
+        buttonClasses={'iu-mb-200'}
+        tooltip={getTooltip()}
+        disabled={getDropdownButtonBeDisabled()}
+        onClick={handleToggleDropdownButtonClick}>
+        Ta hjälp av ICF
+      </CustomButton>
+
+      {shouldRenderDropdown() && (
+        <>
+          <Root ref={rootRef}>
+            <div hidden={!displayDropdown} className={'iu-border-black iu-radius-sm'}>
+              <StyledTitle className={'iu-bg-main iu-color-white iu-p-300'}>
+                <FontAwesomeIcon icon={faInfoCircle} className={'iu-mr-200'} />
+                {modalLabel}
+              </StyledTitle>
+              <ScrollDiv className={'iu-pb-300 iu-bg-white'}>
+                {icfData?.commonCodes.icfCodes && (
+                  <CategoryWrapper className={'iu-bg-white'}>
+                    <p>ICF-kategorier gemensamma för:</p>
+                    <IcfCategory
+                      icfCodeValues={chosenIcfCodeValues}
+                      icdCodes={icfData.commonCodes.icdCodes}
+                      icfCodes={icfData.commonCodes.icfCodes}
+                      onCodeAdd={onCodeAdd}
+                      onCodeRemove={onCodeRemove}
+                    />
+                  </CategoryWrapper>
+                )}
+                {icfData?.uniqueCodes.length &&
+                  icfData.uniqueCodes.map((icfUnique, i) => (
+                    <CategoryWrapper className={'iu-bg-white'} key={i}>
+                      <p>ICF-kategorier för:</p>
+                      <IcfCategory
+                        icfCodeValues={chosenIcfCodeValues}
+                        icdCodes={icfUnique.icdCodes}
+                        icfCodes={icfUnique.icfCodes}
+                        onCodeAdd={onCodeAdd}
+                        onCodeRemove={onCodeRemove}
+                      />
+                    </CategoryWrapper>
+                  ))}
+              </ScrollDiv>
+              <Footer className={'iu-bg-secondary-light iu-p-300'}>
+                <ButtonWrapper>
+                  <CustomButton className={'iu-mr-200'} onClick={handleToggleDropdownButtonClick}>
+                    Stäng
+                  </CustomButton>
+                </ButtonWrapper>
+                <a href={'https://www.socialstyrelsen.se/utveckla-verksamhet/e-halsa/klassificering-och-koder/icf'}>
+                  Läs mer om ICF hos Socialstyrelsenlaunch
+                </a>
+              </Footer>
+            </div>
+          </Root>
+        </>
+      )}
+      {shouldRenderValues() && (
+        <div className={'iu-p-300 iu-bg-grey-200 iu-mt-200 iu-mb-300'}>
+          <p className={'iu-mb-200'}>{collectionsLabel}</p>
+          <ValuesWrapper>
+            {chosenIcfCodeValues?.map((code) => (
+              <p key={code} className={'iu-bg-white iu-p-200 iu-fs-200 iu-radius-sm iu-border-black'}>
+                {code}
+              </p>
+            ))}
+          </ValuesWrapper>
+        </div>
+      )}
+    </>
+  )
+}
+
```

Comment:
See if you can refactor something between 76 and 150. It's a little bit hard to understand the code.